src/Fable.Transforms/Replacements.fs

```diff
@@ -2707,17 +2707,30 @@ let asyncs com (ctx: Context) r t (i: CallInfo) (_: Expr option) (args: Expr lis
     // Fable.Core extensions
     | meth -> Helper.CoreCall("Async", Naming.lowerFirst meth, t, args, i.SignatureArgTypes, ?loc=r) |> Some
 
-let guids (_: ICompiler) (ctx: Context) (_: SourceLocation option) t (i: CallInfo) (thisArg: Expr option) (args: Expr list) =
+let guids (com: ICompiler) (ctx: Context) (r: SourceLocation option) t (i: CallInfo) (thisArg: Expr option) (args: Expr list) =
     match i.CompiledName with
-    | "NewGuid"     -> Helper.CoreCall("String", "newGuid", t, []) |> Some
-    | "Parse"       -> Helper.CoreCall("String", "validateGuid", t, args, i.SignatureArgTypes) |> Some
-    | "TryParse"    -> Helper.CoreCall("String", "validateGuid", t, [args.Head; makeBoolConst true], [args.Head.Type; Boolean]) |> Some
-    | "ToByteArray" -> Helper.CoreCall("String", "guidToArray", t, [thisArg.Value], [thisArg.Value.Type]) |> Some
+    | "NewGuid"     -> Helper.CoreCall("Guid", "newGuid", t, []) |> Some
+    | "Parse"       -> Helper.CoreCall("Guid", "validateGuid", t, args, i.SignatureArgTypes) |> Some
+    | "TryParse"    -> Helper.CoreCall("Guid", "validateGuid", t, [args.Head; makeBoolConst true], [args.Head.Type; Boolean]) |> Some
+    | "ToByteArray" -> Helper.CoreCall("Guid", "guidToArray", t, [thisArg.Value], [thisArg.Value.Type]) |> Some
+    | "ToString" when (args.Length = 0) -> thisArg.Value |> Some
+    | "ToString" when (args.Length = 1) ->
+        match args.Head with
+        | Value (StringConstant "N", _)
+        | Value (StringConstant "D", _)
+        | Value (StringConstant "B", _)
+        | Value (StringConstant "P", _)
+        | Value (StringConstant "X", _) ->
```

Comment:
I would leave this check to the runtime as it may happen the user passes a reference instead of a string literal argument.

Reply:
I combined this with your other comment so that we can still check as much as possible at compile while accounting for references. I figured as few runtime errors as possible is best, let me know if you think I should still change it.